staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go

```diff
@@ -763,6 +712,101 @@ See https://kubernetes.io/docs/reference/using-api/server-side-apply/#conflicts`
 	return nil
 }
 
+func (o *ApplyOptions) serverSideApplyOneObject(info *resource.Info, helper *resource.Helper, data []byte) error {
```

Comment:
FYI for other reviewers, I did a sanity check diff on this extracted `serverSideApplyOneObject` vs the contents of the previous `if o.ServerSideApply {` block. There are only two differences:

```diff
< 		data, err := runtime.Encode(unstructured.UnstructuredJSONScheme, info.Object)
< 		if err != nil {
< 			return cmdutil.AddSourceToErr("serverside-apply", info.Source, err)
---
> 	if o.FieldManager == fieldManagerUnset {
> 		helper = helper.WithFieldManager(fieldManagerServerSideApply)
> 	} else {
> 		helper = helper.WithFieldManager(o.FieldManager)
```

The former was moved out of the block, because auto may need that data untouched as well. The source was also changed to "apply", since we no longer know at that point which flow the data will be used in.

The latter is added below because we can't know until the final per-object operation (i.e. here for SSA) which default manager name should be applied.

Reply:
Thanks for doing the comparison!

It can be helpful to break up a separate commit (or even a separate PR) to make this obvious, but I don't think it's needed here.